src/core/viz/expressions/torque.js

```diff
@@ -193,7 +197,20 @@ export class Torque extends BaseExpression {
         checkFeatureIndependent('torque', 'duration', 1, this.duration);
 
 
-        this.inlineMaker = (inline) =>
-            `(1.- clamp(abs(${inline._input}-${inline._cycle})*(${inline.duration})/(${inline._input}>${inline._cycle}? ${inline.fade.in}: ${inline.fade.out}), 0.,1.) )`;
+        this.preface = `
+        #ifndef TORQUE
+        #define TORQUE
+        float torque(float _input, float cycle, float duration, float fadeIn, float fadeOut){
+            float x = 0.;
+            // Check for NaN
+            if (_input <= 0.0 || 0.0 <= _input){
+                x = 1.- clamp(abs(_input-cycle)*duration/(_input>cycle? fadeIn: fadeOut), 0.,1.);
+            }
+            return x;
```

Comment:
So... now NaN in the input will produce a NaN result in torque, which will not match any value... basically ignoring the input, right?

Reply:
Hmu... No, a NaN in the input will result in a `0` in torque. See that `x` is initialized to `0`

Reply:
Oh, I see. so 0 being a non-match it makes sense